File: typescript/src/models/cardNumberType.ts
// this file is @generated

export enum CardNumberType {
  Fpan = "fpan",
  NetworkToken = "network_token",
}

export const CardNumberTypeSerializer = {
  _fromJsonObject(object: any): CardNumberType {
    return object;
  },

  _toJsonObject(self: CardNumberType): any {
    return self;
  },
};

File: rust/src/models/card_number_type.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
pub enum CardNumberType {
    #[serde(rename = "fpan")]
    Fpan,
    #[serde(rename = "network_token")]
    NetworkToken,
}

impl fmt::Display for CardNumberType {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::Fpan => "fpan",
            Self::NetworkToken => "network_token",
        };
        f.write_str(value)
    }
}
